### PR TITLE
[Tree View] Polish the default design & revise the simple version pages

### PR DESCRIPTION
**Pull request by danilo-leal**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

As the title says! This PR hopefully closes https://github.com/mui/mui-x/issues/11528, and it adds a small coat of polish to the default Tree View look & feel as well as a general revision to the Overview page, applying the latest page structure as described in https://github.com/mui/material-ui/issues/39702. It was a hard one, particularly the disabled section and explaining the `ContentProps` stuff (which felt too obscure).

https://deploy-preview-11529--material-ui-x.netlify.app/x/react-tree-view/

### CODE REVIEW

**Comment by flaviendelangle** [comment]
>  It was a hard one, particularly the disabled section and explaining the ContentProps stuff (which felt too obscure).

I'd like to rework this API which is not consistent with the rest of the X components (they are using slots).

**Comment by flaviendelangle** [comment]
>  Is there a plan to eventually expand this content into multiple pages?

The doc has been split into several pages in #11059 which has just been merged
I first migrated the doc from the core without modifying it and now we are enriching it and restructuring it :+1:

**Comment by danilo-leal** [comment]
@samuelsycamore — that makes sense! I merged this PR with master and pulled all the changes from Flavien's PR linked above. For the sake of not exploding this PR's scope even more, I have constrained myself to tackle only the Simple Tree View pages, also because they mostly house the content that was in the all-encompassing Overview page before. :) We can definitely work on revising the Rich-related pages afterward! 🤙

**Comment by flaviendelangle** [comment]
> We can definitely work on revising the Rich-related pages afterward! 

A lot of it it purely duplicate from `SimpleTreeView` so it should be an easy follow up :+1:

**Comment by flaviendelangle** [comment]
IMHO the main benefit of the `Box` is on small demos, it allow to remove the `sx` prop from the preview
And the `TreeView` is often close to the max length of the preview, so saving 1-2 lines helps a lot.

But for more complex demos which don't have preview anyway, I do agree they don't bring any value.

**Comment by danilo-leal** [comment]
@flaviendelangle I'm assuming your last comment was more about @noraleonte's PR about the Tree View than this one, right? 😅 I just double-checked the changes I'm proposing and didn't find any `sx` prop added directly to the Tree View! :)

**Comment by flaviendelangle** [comment]
Yes my bad, I commented on her PR yesterday, you can forget that one :+1:

**Comment by flaviendelangle** [comment]
:laughing: Polishing the Tree View doc is an hard task with Git conflict.
Respect @danilo-leal :see_no_evil:

**Comment by danilo-leal** [comment]
🤣 I'm considering closing this PR and starting fresh from scratch, heh

**Comment by flaviendelangle** [comment]
The doc structure should be pretty stable now :+1: 
We are mostly adding new doc sections

**Comment by danilo-leal** [comment]
I think this PR is good-to-go for a review whenever y'all can! 👍

File: docs/data/tree-view/overview/overview.md [hunk 1]
```diff
@@ -9,31 +9,26 @@ packageName: '@mui/x-tree-view'
 
 # MUI X Tree View
 
-<p class="description">A Tree View widget presents a hierarchical list.</p>
-
-Tree views can be used to represent a file system navigator displaying folders and files, an item representing a folder can be expanded to reveal the contents of the folder, which may be files, folders, or both.
+<p class="description">The Tree View component displays a hierarchical list of content where you can expand or collapse each parent or child item.</p>
```
[the remark this answers is not in the document]
Comment (danilo-leal) [reply docs/data/tree-view/overview/overview.md]: Great call, yeah!

File: docs/data/tree-view/overview/overview.md [hunk 2]
```diff
@@ -9,31 +9,26 @@ packageName: '@mui/x-tree-view'
...
 {{"component": "modules/components/ComponentLinkHeader.js"}}
 
 ## Available components
 
-There are two versions of the Tree View available.
+The MUI X Tree View package exposes two different versions of the component:
 
-### SimpleTreeView
+- `@mui/x-tree-view/TreeView`: for simpler use cases.
+- `@mui/x-tree-view/RichTreeView`: for more complex and larger uses of the Tree View component.
```
[the remark this answers is not in the document]
Comment (danilo-leal) [reply docs/data/tree-view/overview/overview.md:21]: I actually think we can cut this part altogether given the following headings explain exactly that!

File: docs/data/tree-view/simple-tree-view/selection/selection.md [hunk 3]
```diff
@@ -25,9 +25,8 @@ Use the `disableSelection` prop if you don't want your items to be selectable:
 
 ## Controlled selection
 
-Use the `selectedNodes` prop to control the selected items.
-
-You can use the `onSelectedNodesChange` prop to listen to changes in the selected items and update the prop accordingly.
+Use the `selected` prop to control selected Tree View items.
+You can also use the `onNodeSelect` prop to listen to changes in the selected items and update the prop accordingly.
```
Comment (noraleonte) [inline docs/data/tree-view/simple-tree-view/selection/selection.md:29]:
```suggestion
Use the `selectedNodes` prop to control selected Tree View items.
You can also use the `onSelectedNodesChange` prop to listen to changes in the selected items and update the prop accordingly.
```

File: docs/data/tree-view/simple-tree-view/expansion/expansion.md [hunk 4]
```diff
@@ -9,13 +9,12 @@ waiAria: https://www.w3.org/WAI/ARIA/apg/patterns/treeview/
...
-<p class="description">Handle how users can expand items.</p>
+<p class="description">Learn how to handle expanding and collapsing Tree View items.</p>
 
 ## Controlled expansion
 
-Use the `expandedNodes` prop to control the expanded items.
-
-You can use the `onExpandedNodesChange` prop to listen to changes in the expanded items and update the prop accordingly.
+Use the `expanded` prop to control the expanded items.
+You can also use the `onNodeToggle` prop to listen to changes in the expanded items and update the prop accordingly.
```
Comment (noraleonte) [inline docs/data/tree-view/simple-tree-view/expansion/expansion.md:17]:
```suggestion
Use the `expandedNodes` prop to control the expanded items.
You can also use the `onExpandedNodesChange` prop to listen to changes in the expanded items and update the prop accordingly.
```

File: docs/data/tree-view/simple-tree-view/selection/selection.md
Comment (noraleonte) [inline docs/data/tree-view/simple-tree-view/selection/selection.md:1]: We have very similar docs for the Rich Tree View as well. Maybe it's worth updating the copy there too? 🤔

**Review by noraleonte — COMMENTED** [review]
Thank you for this effort! 🎉 

Leaving some small suggestions for improvements 😄

**Comment by danilo-leal** [comment]
Little bump on this one! :)

**Review by noraleonte — APPROVED** [review]
Looking great! Thanks for picking this up! 🎉 🍰

**Comment by oliviertassinari** [comment]
Nice 👍 

---

There are a few demo heights to update, e.g. 



https://deploy-preview-11529--material-ui-x.netlify.app/x/react-tree-view/simple-tree-view/items/#basics

**Comment by danilo-leal** [comment]
Follow-up PR opened above → https://github.com/mui/mui-x/pull/11979

**Pull request merged**